src/Storages/StorageView.cpp

```diff
@@ -99,10 +130,17 @@ void StorageView::read(
     query_plan.addStep(std::move(materializing));
 
     /// And also convert to expected structure.
-    auto header = metadata_snapshot->getSampleBlockForColumns(column_names, getVirtuals(), getStorageID());
+    const auto & expected_header = metadata_snapshot->getSampleBlockForColumns(column_names, getVirtuals(), getStorageID());
+    const auto & header = query_plan.getCurrentDataStream().header;
+
+    if (changedNullabilityOneWay(header, expected_header))
+    {
+        throw DB::Exception(ErrorCodes::INCORRECT_QUERY,
+                            "Joined columns is nullable, but setting `join_use_nulls` wans't set on CREATE VIEW");
```

Comment:
I think this message can be a little bit better.
Like `Query from view {} returned Nullable column having not Nullable type in structure. If query from view has JOIN, it may be cause by different values of 'json_use_nulls' setting. You may explicitly specify 'json_use_nulls' in 'CREATE VIEW' query to avoid this error.`
Also, we may indeed check if query has JOIN in AST before this check

Reply:
[`76ca408` (#27227)](https://github.com/ClickHouse/ClickHouse/pull/27227/commits/76ca40857a7072ef8303641365ee50829d5df7b9)